frontend/app/helpers/spree/addresses_helper.rb

```diff
@@ -65,9 +65,18 @@ def address_state(form, country, address_id = 'b')
     end
 
     def user_available_addresses
-      return unless try_spree_current_user
+      return [] unless try_spree_current_user
 
-      try_spree_current_user.addresses.where(country: available_countries)
+      checkout_countries = current_store.countries_available_for_checkout
+
+      states = checkout_countries.each_with_object([]) do |country, memo|
```

Comment:
```suggestion
      states = current_store.countries_available_for_checkout.each_with_object([]) do |country, memo|
```
just wanna suggest, less code is better :)